OpenDreamClient/Interface/Prompts/ColorPrompt.cs

```diff
@@ -0,0 +1,184 @@
+using System.Linq;
+using Linguini.Bundle.Errors;
+using OpenDreamShared.Dream;
+using Robust.Client.Graphics;
+using Robust.Client.ResourceManagement;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.Stylesheets;
+
+namespace OpenDreamClient.Interface.Prompts;
+
+internal sealed class ColorPrompt : InputWindow {
+    private readonly BoxContainer _baseControl;
+    private readonly ColorSelectorSliders _colorSelector;
+    private readonly LineEdit _hexColor;
+    private readonly Button _preview;
+    private readonly bool _nullable;
+    private readonly Color _originalColor;
+
+    public ColorPrompt(string title, string message, string defaultValue, bool canCancel,
+        Action<DMValueType, object?>? onClose, bool alpha = false) : base(title, message, true, onClose) {
```

Comment:
Why is this passing `true` instead of `canCancel` to the base? Do color prompts always have a cancel button in byond?

Reply:
Unsure. It was early morning, I might've not been thinking the best haha, but my logic was "if you cancel, it just returns the default color instead", bc I was scared of some edge-case I couldn't remember

Reply:
Color prompts do not always have a cancel button.

`input(usr) as color`:
![image](https://github.com/OpenDreamProject/OpenDream/assets/30789242/55746d5f-1288-4db0-b03f-22d4ae9d2671)

`input(usr) as color|null`:
![image](https://github.com/OpenDreamProject/OpenDream/assets/30789242/39547915-08c5-4422-9301-17e6b336f3d8)

Reply:
Alright, thanks.